File: GeneratedEntityReaderClasses/Components/VerletWeaponComponent.cs
namespace GeneratedNoitaClasses;
public class VerletWeaponComponent : NoitaComponentBase,  iNoitaType<VerletWeaponComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat damage_radius { get; set; } = new();
	public NoitaFloat physics_force_radius { get; set; } = new();
	public NoitaFloat damage_min_step { get; set; } = new();
	public NoitaFloat damage_max { get; set; } = new();
	public NoitaFloat damage_coeff { get; set; } = new();
	public NoitaFloat impulse_coeff { get; set; } = new();
	public NoitaInt fade_duration_frames { get; set; } = new();
	public NoitaFloat physics_impulse_coeff { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		damage_radius.Read(s);
		physics_force_radius.Read(s);
		damage_min_step.Read(s);
		damage_max.Read(s);
		damage_coeff.Read(s);
		impulse_coeff.Read(s);
		fade_duration_frames.Read(s);
		physics_impulse_coeff.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		damage_radius.Write(s);
		physics_force_radius.Write(s);
		damage_min_step.Write(s);
		damage_max.Write(s);
		damage_coeff.Write(s);
		impulse_coeff.Write(s);
		fade_duration_frames.Write(s);
		physics_impulse_coeff.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/VerletWeaponComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class VerletWeaponComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    damage_radius:NoitaFloat = field(default_factory=NoitaFloat)
    physics_force_radius:NoitaFloat = field(default_factory=NoitaFloat)
    damage_min_step:NoitaFloat = field(default_factory=NoitaFloat)
    damage_max:NoitaFloat = field(default_factory=NoitaFloat)
    damage_coeff:NoitaFloat = field(default_factory=NoitaFloat)
    impulse_coeff:NoitaFloat = field(default_factory=NoitaFloat)
    fade_duration_frames:NoitaInt = field(default_factory=NoitaInt)
    physics_impulse_coeff:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.damage_radius.Read(s)
        self.physics_force_radius.Read(s)
        self.damage_min_step.Read(s)
        self.damage_max.Read(s)
        self.damage_coeff.Read(s)
        self.impulse_coeff.Read(s)
        self.fade_duration_frames.Read(s)
        self.physics_impulse_coeff.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.damage_radius.Write(s)
        self.physics_force_radius.Write(s)
        self.damage_min_step.Write(s)
        self.damage_max.Write(s)
        self.damage_coeff.Write(s)
        self.impulse_coeff.Write(s)
        self.fade_duration_frames.Write(s)
        self.physics_impulse_coeff.Write(s)

